File: src/main/java/xyz/mlhmz/financemanager/entities/Sheet.java
package xyz.mlhmz.financemanager.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.UuidGenerator;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
public class Sheet {
    @Id
    @UuidGenerator
    private UUID uuid;

    private String title;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    @OneToMany(cascade = {CascadeType.PERSIST, CascadeType.REMOVE})
    private List<Transaction> transactions;

    @ManyToOne(optional = false)
    private User user;
}

File: frontend/src/sheets/Sheet.ts
import { z } from "zod";
import type { Transaction } from "../transactions/Transaction";

export interface Sheet {
	uuid?: string;
	title?: string;
	createdAt?: string;
	updatedAt?: string;
	transactions?: Transaction[];
}

export interface SheetStats {
	sum?: number;
}

export const MutateSheet = z.object({
	title: z.string(),
});

export type MutateSheet = z.infer<typeof MutateSheet>;
